mlir/lib/Dialect/Rock/Tuning/ConvContext.cpp

```diff
@@ -23,15 +23,26 @@ populateDimIndexAndSize(const ArrayAttr &layoutAttr,
   size_t dimValSize = layoutAttr.size();
   for (size_t i = 0; i < dimValSize; ++i) {
     auto key = layoutAttr.getValue()[i].cast<StringAttr>().getValue();
+
+    // +++pf: update old keys.
+    if (key == "y") key = "0";
+    if (key == "x") key = "1";
+    if (key[0] == 'h')
+      key = StringAttr::get(layoutAttr.getContext(),
```

Comment:
It is a pain in the butt to take a string attribute and replace a character.

Reply:
You shouldn't need to `std::string` here. `Twine()` should do you?